client-mr/src/main/java/org/apache/hadoop/mapreduce/RssMRUtils.java

```diff
@@ -38,19 +38,29 @@
 public class RssMRUtils {
 
   private static final Logger LOG = LoggerFactory.getLogger(RssMRUtils.class);
+  private static int MAX_TASK_LENGTH = 19;
+  private static int MAX_ATTEMPT_LENGTH = 2;
+  private static long MAX_TASK_ID = (1 << MAX_TASK_LENGTH) - 1;
+  private static long MAX_ATTEMPT_ID = (1 << MAX_ATTEMPT_LENGTH) - 1;
 
-  // Class TaskAttemptId have two field id and mapId, mapId is 4 low byte,
-  // id is high 4 byte.
+  // Class TaskAttemptId have two field id and mapId, rss taskAttemptID have 21 bits,
+  // mapId is 19 bits, id is 2 bits.
```

Comment:
Is 2-bit for attemptID is too tricky ?

Reply:
When we increate a bit, we will increate double meta memory.